### PR TITLE
[DRAFT] My original fine grained solution

### PR DESCRIPTION
**Pull request by bviyer**

### CODE REVIEW

**Comment by bviyer** [comment]
Here is the test code that I tried it with:

```
odule {
  util.global public @handle : !cudnn.handle
  util.initializer {
    %device = hal.ex.shared_device : !hal.device
    %0 = cudnn.handle(%device) : !cudnn.handle
    util.global.store %0, @handle : !cudnn.handle
    util.initializer.return
  }
  cudnn.graph @conv2d(%arg0: !cudnn.tensor<8x32x4x4xf64, NHWC>, %arg1: !cudnn.tensor<32x32x1x1xf64, KHWC>) -> !cudnn.tensor<8x32x4x4xf64, NHWC> {
    %0 = cudnn.convolution(%arg0, %arg1) alpha = 1.000000e+00 beta = 0.000000e+00 spatial_dim_count = 2 spatial_stride = [1, 1] pre_padding = [0, 0] post_padding = [0, 0] dilation = [1, 1] : (!cudnn.tensor<8x32x4x4xf64, NHWC>, !cudnn.tensor<32x32x1x1xf64, KHWC>) -> !cudnn.tensor<8x32x4x4xf64, NHWC>
    cudnn.return %0 : !cudnn.tensor<8x32x4x4xf64, NHWC>
  }
  util.global public @x = dense<1.000000e+00> : tensor<8x4x4x32xf64>
  util.global public @w = dense<1.000000e+00> : tensor<32x1x1x32xf64>
  func.func @main() -> tensor<8x4x4x32xf64> {
    %x = util.global.load @x : tensor<8x4x4x32xf64>
    %w = util.global.load @w : tensor<32x1x1x32xf64>
    %handle = util.global.load @handle : !cudnn.handle
    %0 = cudnn.call handle(%handle) @conv2d(%x, %w) : (tensor<8x4x4x32xf64>, tensor<32x1x1x32xf64>) -> tensor<8x4x4x32xf64>
    return %0 : tensor<8x4x4x32xf64>
  }
}
```

And this is the error I found:

```
within split at /tmp/test.mlir:1 offset :20:5: error: failed to legalize unresolved materialization from 'tensor<8x4x4x32xf64>' to 'tensor<8x4x4x32xf32>' that remained live after conversion
    return %0 : tensor<8x4x4x32xf64>
    ^
within split at /tmp/test.mlir:1 offset :20:5: note: see current operation: %6 = "builtin.unrealized_conversion_cast"(%5) : (tensor<8x4x4x32xf64>) -> tensor<8x4x4x32xf32>
within split at /tmp/test.mlir:1 offset :20:5: note: see existing live user here: "func.return"(%6) : (tensor<8x4x4x32xf32>) -> ()
```

However, I tried to dump the module after the failed(...) funtion and this is what I see:
```
Failed Module: module {
  util.global public @handle : !cudnn.handle
  util.initializer {
    %device = hal.ex.shared_device : !hal.device
    %0 = cudnn.handle(%device) : !cudnn.handle
    util.global.store %0, @handle : !cudnn.handle
    util.initializer.return
  }
  cudnn.graph @conv2d(%arg0: !cudnn.tensor<8x32x4x4xf64, NHWC>, %arg1: !cudnn.tensor<32x32x1x1xf64, KHWC>) -> !cudnn.tensor<8x32x4x4xf64, NHWC> {
    %0 = cudnn.convolution(%arg0, %arg1) alpha = 1.000000e+00 beta = 0.000000e+00 spatial_dim_count = 2 spatial_stride = [1, 1] pre_padding = [0, 0] post_padding = [0, 0] dilation = [1, 1] : (!cudnn.tensor<8x32x4x4xf64, NHWC>, !cudnn.tensor<32x32x1x1xf64, KHWC>) -> !cudnn.tensor<8x32x4x4xf64, NHWC>
    cudnn.return %0 : !cudnn.tensor<8x32x4x4xf64, NHWC>
  }
  util.global public @x = dense<1.000000e+00> : tensor<8x4x4x32xf64>
  util.global public @w = dense<1.000000e+00> : tensor<32x1x1x32xf64>
  func.func @main() -> tensor<8x4x4x32xf64> {
    %x = util.global.load @x : tensor<8x4x4x32xf64>
    %w = util.global.load @w : tensor<32x1x1x32xf64>
    %handle = util.global.load @handle : !cudnn.handle
    %0 = cudnn.call handle(%handle) @conv2d(%x, %w) : (tensor<8x4x4x32xf64>, tensor<32x1x1x32xf64>) -> tensor<8x4x4x32xf64>
    return %0 : tensor<8x4x4x32xf64>
  }
}
```
I am not even seeing the f32 return that the compiler is complaining about. I am not sure how to get to it.


Also, I am sure I am missing something basic. Any help/pointers you have is greatly appreciated.

File: compiler/src/iree/compiler/Dialect/Util/Transforms/ConvertPrimitiveType.cpp
Comment (ScottTodd) [inline compiler/src/iree/compiler/Dialect/Util/Transforms/ConvertPrimitiveType.cpp:1]: Can you update the PR title and description with more context for what you're trying to do/solve? I'm not sure what to review here.

Comment (bviyer) [reply compiler/src/iree/compiler/Dialect/Util/Transforms/ConvertPrimitiveType.cpp:1]: Apologies. This is a draft patch that I am using to discuss an issue. It is not to be checked in.

Comment (ScottTodd) [reply compiler/src/iree/compiler/Dialect/Util/Transforms/ConvertPrimitiveType.cpp:1]: Ah, I see. You can mark a PR as a "draft": https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request that will avoid tagging "code owners". That's why a few reviews were requested: https://github.com/openxla/iree/blob/6db40b30264ee43a9d43bdbd639604d89ad7472d/.github/CODEOWNERS#L14-L16

**Pull request closed without merging**